### PR TITLE
Correct ocean conservation check settings

### PR DESCRIPTION
**Pull request by mark-petersen**
Currently, the ocean conservation check analysis member overwrites the first entry in the file with a zero after restarts for some variables. This PR corrects this behavior so that the first day's entry in a monthly conservation check file is identical between continuous runs and a run with a restart break.

Fixes https://github.com/E3SM-Project/E3SM/issues/6642

[NML] for some mpaso resolutions
[BFB] for all tested files

### CODE REVIEW

File: components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml [hunk 1]
```diff
@@ -1231,38 +1231,11 @@
...
-<config_AM_conservationCheck_enable ocn_grid="SOwISC12to60E2r4">.true.</config_AM_conservationCheck_enable>
-<config_AM_conservationCheck_enable ocn_grid="ECwISC30to60E2r1">.true.</config_AM_conservationCheck_enable>
-<config_AM_conservationCheck_enable ocn_grid="IcoswISC30E3r5">.true.</config_AM_conservationCheck_enable>
-<config_AM_conservationCheck_enable ocn_grid="IcosXISC30E3r7">.true.</config_AM_conservationCheck_enable>
-<config_AM_conservationCheck_enable ocn_grid="FRISwISC08to60E3r1">.true.</config_AM_conservationCheck_enable>
-<config_AM_conservationCheck_enable ocn_grid="FRISwISC04to60E3r1">.true.</config_AM_conservationCheck_enable>
-<config_AM_conservationCheck_enable ocn_grid="FRISwISC02to60E3r1">.true.</config_AM_conservationCheck_enable>
-<config_AM_conservationCheck_enable ocn_grid="FRISwISC01to60E3r1">.true.</config_AM_conservationCheck_enable>
-<config_AM_conservationCheck_enable ocn_grid="RRSwISC6to18E3r5">.true.</config_AM_conservationCheck_enable>
+<config_AM_conservationCheck_enable>.true.</config_AM_conservationCheck_enable>
```
Comment (xylar) [inline components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml:1234]: @mark-petersen, we may want this turned off for QU240, etc.   For light-weight testing.

File: components/mpas-seaice/bld/namelist_files/namelist_defaults_mpassi.xml [hunk 2]
```diff
@@ -512,7 +512,7 @@
 <config_AM_ridgingDiagnostics_write_on_startup>true</config_AM_ridgingDiagnostics_write_on_startup>
 
 <!-- AM_conservationCheck -->
-<config_AM_conservationCheck_enable>false</config_AM_conservationCheck_enable>
+<config_AM_conservationCheck_enable>true</config_AM_conservationCheck_enable>
```
Comment (xylar) [inline components/mpas-seaice/bld/namelist_files/namelist_defaults_mpassi.xml:515]: This should be a separate PR, I think.

Comment (jonbob) [reply components/mpas-seaice/bld/namelist_files/namelist_defaults_mpassi.xml:515]: Thanks @xylar -- I agree, especially since mpas-seaice is also used for F-cases where its performance is already an issue

File: components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml (hunk 1 above)
Comment (xylar) [reply components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml:1234]: Let's explicitly set this to `.false.` for:
* `oQU240`
* `oQU480`
* `oQU240wLI`
* `oQU120`

These meshes are used for testing:
https://github.com/E3SM-Project/E3SM/blob/master/cime_config/tests.py
and we don't want to needlessly increase test burden.

**Comment by proteanplanet** [comment]
Conservation checks in both the sea ice nor ocean models should not be set as a default. It was actually a mistake that they were left on in the ocean for V3 production runs - they should have been switched off.  They add significant computational expense - up to 30% for the sea ice model alone.  Please remove the lines in [components/mpas-seaice/bld/namelist_files/namelist_defaults_mpassi.xml](https://github.com/E3SM-Project/E3SM/pull/6643/files/29ff9d05bc701c5ff4c52d9f32dfd5da0724acf9#diff-d9c675d17b8538db7faf430dada4c4d0635a845c4a04dc578fd082e7a7240a84) from this PR.   Also please consider removing this as default behavior for the ocean.  Conservation checks should be used for development purposes, but once we know models are conserving after checks run for PRs, the analysis members should be kept off.

**Comment by xylar** [comment]
@mark-petersen, in light of @proteanplanet's comment above, which makes sense to me, can you actually turn the conservation analysis member off by default for all meshes in this PR.  I think we don't want to get distracted form the main objective here, which was to fix lack of restart capability in the conservation analysis member.

**Comment by rljacob** [comment]
Could we add a restart test that does turn on the conservation analysis?

**Comment by xylar** [comment]
@rljacob, that's an excellent idea but will CIME notice whether the output is BFB here?  I seem to recall that ERS tests don't include output from analysis, but maybe I'm misremembering.

**Comment by jonbob** [comment]
@rljacob -- we can do that, though I think mpas analysis member output is ignored in the tests

**Comment by xylar** [comment]
@mark-petersen, are you comfortable adding a test with the conservation check on or would you like me to do it?

**Comment by mark-petersen** [comment]
I made conservation checks default off for both sea ice and ocean, per comment by @proteanplanet above. I'm checking the restart capability now.

File: components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml [hunk 3]
```diff
@@ -1232,37 +1232,10 @@
...
-<config_AM_conservationCheck_write_on_startup ocn_grid="SOwISC12to60E2r4">.true.</config_AM_conservationCheck_write_on_startup>
-<config_AM_conservationCheck_write_on_startup ocn_grid="ECwISC30to60E2r1">.true.</config_AM_conservationCheck_write_on_startup>
-<config_AM_conservationCheck_write_on_startup ocn_grid="IcoswISC30E3r5">.true.</config_AM_conservationCheck_write_on_startup>
-<config_AM_conservationCheck_write_on_startup ocn_grid="IcosXISC30E3r7">.true.</config_AM_conservationCheck_write_on_startup>
-<config_AM_conservationCheck_write_on_startup ocn_grid="FRISwISC08to60E3r1">.true.</config_AM_conservationCheck_write_on_startup>
-<config_AM_conservationCheck_write_on_startup ocn_grid="FRISwISC04to60E3r1">.true.</config_AM_conservationCheck_write_on_startup>
-<config_AM_conservationCheck_write_on_startup ocn_grid="FRISwISC02to60E3r1">.true.</config_AM_conservationCheck_write_on_startup>
-<config_AM_conservationCheck_write_on_startup ocn_grid="FRISwISC01to60E3r1">.true.</config_AM_conservationCheck_write_on_startup>
-<config_AM_conservationCheck_write_on_startup ocn_grid="RRSwISC6to18E3r5">.true.</config_AM_conservationCheck_write_on_startup>
```
Comment (xylar) [inline components/mpas-ocean/bld/namelist_files/namelist_defaults_mpaso.xml:1265]: I agree that we want all these changes based on my comment https://github.com/E3SM-Project/E3SM/issues/6642#issuecomment-2373411357

File: components/mpas-ocean/cime_config/buildnml [hunk 4]
```diff
@@ -1229,9 +1229,9 @@ def buildnml(case, caseroot, compname):
             lines.append('        filename_interval="output_interval"')
             lines.append('        clobber_mode="truncate"')
             lines.append('        packages="conservationCheckAMPKG"')
-            lines.append('        input_interval="none"')
+            lines.append('        input_interval="initial_only"')
             lines.append('        reference_time="01-01-01_00:00:00"')
-            lines.append('        output_interval="none">')
+            lines.append('        output_interval="stream:restart:output_interval">')
```
Comment (xylar) [inline components/mpas-ocean/cime_config/buildnml:1234]: Let's revert these changes.  I didn't understand that restarts are not needed because we are purposefully starting the average over at every output interval (i.e. accumulating only for 1 day).

Instead, let's change to `clobber_mode="append"` above in `conservationCheckOutput`.

**Comment by xylar** [comment]
@mark-petersen, I've got a version of your branch with my suggested modifications here:
https://github.com/xylar/E3SM/commits/mark-petersen/mpas/correct-conservation-check/
It includes a new stealth test.  I have included MPAS-Ocean output files in that test by using `sed` to modify the `env_archive.xml` file.  Not sure that's kosher but I couldn't find a better way.

**Comment by mark-petersen** [comment]
Updated using @xylar corrections based on previous comment and [comment on issue page](https://github.com/E3SM-Project/E3SM/issues/6642#issuecomment-2373411357). Thanks! Retesting now...

File: components/mpas-ocean/cime_config/testdefs/testmods_dirs/mpaso/conservation_check/shell_commands [hunk 5]
```diff
@@ -0,0 +1,2 @@
+# include mpas-ocean outputs in testing
+sed -i 's#compclass="ocn" exclude_testing="true"#compclass="ocn" exclude_testing="false"#g' env_archive.xml
```
Comment (xylar) [inline components/mpas-ocean/cime_config/testdefs/testmods_dirs/mpaso/conservation_check/shell_commands:2]: @jonbob and @rljacob, let me know what you think if this approach.  Happy to hear of other suggestions for how to not exclude MPAS-Ocean files from this particular test.

**Comment by mark-petersen** [comment]
Updated the description.

File: cime_config/tests.py [hunk 6]
```diff
@@ -266,6 +266,7 @@
             "SMS_D_Ld1.T62_oQU240wLI.GMPAS-IAF-PISMF.mpaso-impl_top_drag",
             "SMS_D_Ld1.T62_oQU240.GMPAS-IAF.mpaso-harmonic_mean_drag",
             "SMS_D_Ld1.T62_oQU240.GMPAS-IAF.mpaso-upwind_advection",
+            "ERS_D.T62_oQU240.GMPAS-IAF.mpaso-conservation_check",
             )
```
Comment (jonbob) [inline cime_config/tests.py:270]: could this be ERS_Ld5_D instead? Most of the stealth tests are short because they get run on the cloud

Comment (xylar) [reply cime_config/tests.py:270]: Works for me.  @mark-petersen, would you make that change?

File: components/mpas-ocean/cime_config/testdefs/testmods_dirs/mpaso/conservation_check/shell_commands (hunk 5 above)
Comment (rljacob) [reply components/mpas-ocean/cime_config/testdefs/testmods_dirs/mpaso/conservation_check/shell_commands:2]: Since its an xml file, can xmlchange do that?   If not, sed is fine.

Comment (xylar) [reply components/mpas-ocean/cime_config/testdefs/testmods_dirs/mpaso/conservation_check/shell_commands:2]: I couldn't figure out how.  I think `xmlchange` can only be used to modify specific fields that have an `id` and `value`, and this is not one of those.

**Comment by mark-petersen** [comment]
I can now confirm the expected behavior from this PR.

I ran QU240 with `config_am_conservationcheck_enable = .true.` for all cases, and everything standard from the repo for four cases here `/lcrc/group/e3sm/ac.mpetersen/scratch/chrys/`

1. master, 2 months `240924i.ConsCheckMasterFull.T62_oQU240.GMPAS-IAF.chr/`
2. master 1mo+restart+1 mo `240924j.ConsCheckMasterRestart.T62_oQU240.GMPAS-IAF.chr/`
3. pr 2 months `240924k.ConsCheckPrFull.T62_oQU240.GMPAS-IAF.chr/`
4. pr 1mo+restart+1 mo `240924l.ConsCheckPrRestart.T62_oQU240.GMPAS-IAF.chr/`

writing restarts every month. For reference, I used

```
./create_newcase -case $CASE_ROOT/$E3SM_CASE   -compiler gnu   -mach chrysalis   -project e3sm \
   -compset GMPAS-IAF -res T62_oQU240 --walltime 00:30:00
```

In case 2, the first entry differes from case 1 by being either zero or differing:
```
$ ncdump -v accumulatedRainFlux  *mpaso.hist.am.conservationCheck*01-02-01.nc |tail
 accumulatedRainFlux = 12140753948.2218, 12114183191.7286, 12117918123.2804,  (case 1)
 accumulatedRainFlux = 12129038639.0062, 12114183191.7286, 12117918123.2804, (case 2)
...

$ ncdump -v accumulatedRiverRunoffTemperatureFlux *mpaso.hist.am.conservationCheck*01-02-01.nc | tail
 accumulatedRiverRunoffTemperatureFlux = 18589454.7081383, 18605091.7352952, (case 1)
 accumulatedRiverRunoffTemperatureFlux = 0, 18605091.7352952, (case 2)
```
Running from this PR, case 3 and case 4 have identical output, identical to case 1 above.

**Review by xylar — APPROVED** [review]
I'm approving based on @mark-petersen's testing and my own.

File: components/mpas-ocean/cime_config/testdefs/testmods_dirs/mpaso/conservation_check/shell_commands (hunk 5 above)
Comment (jonbob) [reply components/mpas-ocean/cime_config/testdefs/testmods_dirs/mpaso/conservation_check/shell_commands:2]: I slacked Jim to ask if he knows...

**Comment by jonbob** [comment]
I've been testing with the new testdef and don't believe it's working correctly -- first, it doesn't seem to change "exclude_testing" for mpaso in env_archive.xml. However, even if I change that by hand, it still does not run cprnc on the conservation am file -- it might take some work to get testing to recognize that file

**Comment by xylar** [comment]
@jonbob, that's odd.  I think it changed the file for me.  In:
```
/lcrc/group/e3sm/ac.xylar/scratch/chrys/ERS_D.T62_oQU240.GMPAS-IAF.chrysalis_intel.mpaso-conservation_check.20240925_060737_57gegd
```
I'm seeing:
```
    <comp_archive_spec compname="mpaso" compclass="ocn" exclude_testing="false">
      <rest_file_extension>rst</rest_file_extension>
...
```
as expected.

As for the latter point.  I wasn't sure how to find out what cprnc is checking.  I guess we'll need @jgfouca's help to figure out another way.

**Comment by jonbob** [comment]
@xylar -- that's very odd... Anyway, Jim didn't see a way to do it with xmlchange either.

**Comment by xylar** [comment]
@jonbob, I'll try it again.  In the meantime, can you clarify what you are doing to see what files cprnc is looking at?

**Comment by xylar** [comment]
I reset to the current state of this branch and I'm also seeing that the `sed` changes aren't happening -- so odd!

**Comment by xylar** [comment]
My fault.  I accidentally had e3sm-unified loaded when I launched the test.  It must have a better `sed` that supports `-i` but the default `sed` on Chrysalis doesn't (!!!).

I'll have to do this more manually.

**Comment by jonbob** [comment]
@xylar -- there are files that get made for tests -- TestStatus and TestStatus.log. If I grep TestStatus.log for cprnc, I only see that being run on the cpl.hi files

**Comment by xylar** [comment]
Thanks @jonbob.  I'll try to debug some more tomorrow.

**Comment by xylar** [comment]
On second thought, I'm beginning to feel like getting an ERS test to work with MPAS-Ocean history output is too big a project to be in the scope of this PR.

@jonbob and @rljacob, I think the only plausible fallback is to take out the stealth test.  What about you?

**Comment by jonbob** [comment]
@xylar -- I was going to recommend the same thing. I think it's more important to fix the functionality, and we can add a test once we get it working

**Comment by rljacob** [comment]
Don't try to do the sed magic and force the comparison but leave on the test that enables the conservation analysis members.

**Comment by xylar** [comment]
@rljacob, okay, good suggestion.  I have noted in the README for the test that it does not currently include MPAS-Ocean history files.  I have given a manual change that folks could make to `cime_config/config_archive.xml` if they want to include these files.

File: components/mpas-ocean/cime_config/testdefs/testmods_dirs/mpaso/conservation_check/README [hunk 7]
```diff
@@ -0,0 +1,10 @@
+This testdef is used to test the conservation check analysis member, which
+wasintroduced in MPAS-Ocean PR #4521 and has been made a stealth feature in
+#6643. This test turns on the consevation check analysis member by setting:
+
+    config_AM_conservationCheck_enable = .true.
+
+However, it shoudl be noted that MPAS-Ocean history files are not currently
+included in E3SM testing so non-BFB results will not be detected unless one
+manually changes to 'compname="mpaso" exclude_testing="false"' in the file
+cime_config/config_archive.xml.
```
Comment (jonbob) [inline components/mpas-ocean/cime_config/testdefs/testmods_dirs/mpaso/conservation_check/README:10]: one quick comment -- even changing exclude_testing won't check any of the mpaso history files. Plus there's a minor misspelling: "shoudl"

Comment (xylar) [reply components/mpas-ocean/cime_config/testdefs/testmods_dirs/mpaso/conservation_check/README]: @jonbob, okay, I understood that it would but I must be mistaken.  I have removed this text and fixed the typo.

Comment (xylar) [reply components/mpas-ocean/cime_config/testdefs/testmods_dirs/mpaso/conservation_check/README]: Okay, clearly a bigger can of worms that remains beyond the scope of this PR.  Thanks for being more thorough than I was and for keeping things honest.

**Comment by jonbob** [comment]
@xylar -- I tested this in the shell_commands and it does give us what we want:
````
sed -i 's/\(compname="mpaso" exclude_testing="true"\)/compname="mpaso"/' env_archive.xml
sed -i '/\(compname="mpaso"\)/,/<hist_file_extension>/ s/<hist_file_extension>hist/<hist_file_extension>hist.am.conservationCheck\\..*\\.nc$/' env_archive.xml
````
I'm not convinced we need to make the test completely functional for this PR, but I was curious about how difficult it would be...

**Review by darincomeau — APPROVED** [review]
Approve the changes to the conservation check based on other's testing and visual inspection.
I'll defer to others on the added testing.

**Comment by jonbob** [comment]
Passes:
* ERP_Ld3.ne30pg2_r05_IcoswISC30E3r5.WCYCL1850.chrysalis_intel.allactive-pioroot1 (with expected NML DIFF)
* ERS_Ld5.T62_oQU120.CMPASO-NYF.chrysalis_intel

merged to next

**Comment by jonbob** [comment]
merged to master and expected NML DIFFs blessed
